File: push/src/main/java/com/xiaomi/channel/commonutils/misc/BuildSettings.java
package com.xiaomi.channel.commonutils.misc;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/misc/BuildSettings.class */
public class BuildSettings {
    public static final String BETA = "BETA";
    public static final String DEBUG = "DEBUG";
    public static final boolean IsBetaBuild;
    public static final boolean IsDebugBuild;
    public static final boolean IsDefaultChannel;
    public static final boolean IsForYYBuild;
    public static final boolean IsLogableBuild;
    public static final boolean IsRCBuild;
    public static boolean IsTestBuild = false;
    public static final String LOGABLE = "LOGABLE";
    public static final int Official = 1;
    public static final int OneBox = 3;
    public static final String ReleaseChannel;
    public static final int SandBox = 2;
    public static final String TEST = "TEST";
    public static final String YY = "YY";
    private static int envType;

    static {
        String str = DebugSwitch.sDebugServerHost ? "ONEBOX" : "@SHIP.TO.2A2FE0D7@";
        ReleaseChannel = str;
        boolean zContains = str.contains("2A2FE0D7");
        IsDefaultChannel = zContains;
        IsDebugBuild = zContains || DEBUG.equalsIgnoreCase(str);
        IsLogableBuild = LOGABLE.equalsIgnoreCase(str);
        IsForYYBuild = str.contains(YY);
        IsTestBuild = str.equalsIgnoreCase(TEST);
        IsBetaBuild = BETA.equalsIgnoreCase(str);
        boolean z = false;
        if (str.startsWith("RC")) {
            z = true;
        }
        IsRCBuild = z;
        envType = 1;
        if (str.equalsIgnoreCase("SANDBOX")) {
            envType = 2;
        } else if (str.equalsIgnoreCase("ONEBOX")) {
            envType = 3;
        } else {
            envType = 1;
        }
    }

    public static boolean IsOneBoxBuild() {
        return envType == 3;
    }

    public static boolean IsSandBoxBuild() {
        return envType == 2;
    }

    public static int getEnvType() {
        return envType;
    }

    public static void setEnvType(int i) {
        envType = i;
    }
}

File: legacy/src/main/java/com/xiaomi/channel/commonutils/misc/BuildSettings.kt
package com.xiaomi.channel.commonutils.misc

object BuildSettings {
    const val BETA = "BETA"
    const val DEBUG = "DEBUG"
    const val LOGABLE = "LOGABLE"
    const val TEST = "TEST"
    const val YY = "YY"
    const val Official = 1
    const val SandBox = 2
    const val OneBox = 3

    val ReleaseChannel: String
    val IsDefaultChannel: Boolean
    val IsDebugBuild: Boolean
    val IsLogableBuild: Boolean
    val IsForYYBuild: Boolean
    var IsTestBuild: Boolean = false
    val IsBetaBuild: Boolean
    val IsRCBuild: Boolean

    private var envType: Int

    init {
        val str = if (DebugSwitch.sDebugServerHost) "ONEBOX" else "@SHIP.TO.2A2FE0D7@"
        ReleaseChannel = str
        val zContains = str.contains("2A2FE0D7")
        IsDefaultChannel = zContains
        IsDebugBuild = zContains || DEBUG.equals(str, ignoreCase = true)
        IsLogableBuild = LOGABLE.equals(str, ignoreCase = true)
        IsForYYBuild = str.contains(YY)
        IsTestBuild = str.equals(TEST, ignoreCase = true)
        IsBetaBuild = BETA.equals(str, ignoreCase = true)
        IsRCBuild = str.startsWith("RC")
        envType = when {
            str.equals("SANDBOX", ignoreCase = true) -> 2
            str.equals("ONEBOX", ignoreCase = true) -> 3
            else -> 1
        }
    }

    @JvmStatic
    fun IsOneBoxBuild(): Boolean = envType == 3

    @JvmStatic
    fun IsSandBoxBuild(): Boolean = envType == 2

    @JvmStatic
    fun getEnvType(): Int = envType

    @JvmStatic
    fun setEnvType(type: Int) {
        envType = type
    }
}
